include/rocksdb/filter_policy.h

```diff
@@ -212,4 +212,15 @@ class FilterPolicy {
 // trailing spaces in keys.
 extern const FilterPolicy* NewBloomFilterPolicy(
     double bits_per_key, bool use_block_based_builder = false);
+
+// An EXPERIMENTAL new Bloom alternative that saves about 30% space
+// compared to Bloom filters, with about 3-4x construction time and
+// similar query times. For example, if you pass in 10 for
+// bloom_equivalent_bits_per_key, you'll get the same 0.95% FP rate
+// as Bloom filter but only using about 7 bits per key. (This
+// way of configuring the new filter is considered experimental
+// and/or transitional, so is expected to go away.)
+extern const FilterPolicy* NewExperimentalRibbonFilterPolicy(
+    double bloom_equivalent_bits_per_key);
```

Comment:
The interface exposes the filter size (bits_per_key), but also uses `equivalent` of fp to bloom, which I feel might confuse the user.
Maybe we should switch to using fp as interface for long-term, but stick with filter size (bits_per_key) for now, and have a helper function/interface to bridge them?

Reply:
The complication with Ribbon is that if you pin the bits per key, your FP rate varies a little bit based on number of keys in the filter. If you pin the FP rate (what I've done here with Bloom equiv), your Ribbon bits per key varies a little bit based on number of keys in the filter. And consequently, there's no easy translation between the two.

Though that's assuming a relatively fixed construction success probability and (thus) construction CPU overhead. If you let that be more flexible, you *could* have a fixed relationship between FP rate and Ribbon bits per key, but then you'd "pay extra" for scaling to large number of keys per filter. You already have to pay for larger working size (just like building a hash table); you don't want to compound that with lowered construction success probability. (In fact, in this implementation I slightly favor higher-than-advertised construction success probability for large number of keys per filter.)